### PR TITLE
refactor storageProcess args

### PR DESCRIPTION
**Pull request by axenteoctavian**
## Reasoning behind the pull request
- refactor storageProcess args
- 
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: epochStart/bootstrap/process_test.go [hunk 1]
```diff
@@ -673,12 +679,32 @@ func TestNewEpochStartBootstrap_NilArgsChecks(t *testing.T) {
...
+			ShardCoordinatorFactory:          &testscommon.MultiShardCoordinatorFactoryMock{},
+			AdditionalStorageServiceFactory:  &testscommon.AdditionalStorageServiceFactoryMock{},
+			NodesCoordinatorWithRaterFactory: nil,
+		}
 
 		epochStartProvider, err := NewEpochStartBootstrap(args)
 		require.Nil(t, epochStartProvider)
 		require.True(t, errors.Is(err, errorsMx.ErrNilNodesCoordinatorFactory))
 	})
+	t.Run("nil NodesCoordinatorWithRaterFactory should error", func(t *testing.T) {
```
Comment (mariusmihaic) [inline epochStart/bootstrap/process_test.go:692]: Same remark for all these tests as in previous prs

File: epochStart/bootstrap/storageProcess_test.go [hunk 2]
```diff
@@ -64,37 +56,36 @@ func TestNewStorageEpochStartBootstrap_InvalidArgumentsShouldErr(t *testing.T) {
 		sesb, err := NewStorageEpochStartBootstrap(args)
 		assert.True(t, check.IfNil(sesb))
 		assert.True(t, errors.Is(err, epochStart.ErrNilHasher))
-
 	})
-	t.Run("nil ChanGracefullyClose should err", func(t *testing.T) {
+	t.Run("nil RunTypeComponents should err", func(t *testing.T) {
```
Comment (mariusmihaic) [inline epochStart/bootstrap/storageProcess_test.go:60]: Shouldn't we check nil pointers for RequestCreator and NodesCoordCreator? Or are they tested somwhere else?

File: epochStart/mock/requestHandlerFactoryMock.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package mock
+
+import (
+	"github.com/multiversx/mx-chain-go/dataRetriever/requestHandlers"
+	"github.com/multiversx/mx-chain-go/process"
+	"github.com/multiversx/mx-chain-go/testscommon"
+)
+
+// RequestHandlerFactoryMock -
+type RequestHandlerFactoryMock struct {
```
Comment (mariusmihaic) [inline epochStart/mock/requestHandlerFactoryMock.go:10]: This is not being used anywhere + there is already another identical mock like this in `testscommon/factory`

File: epochStart/mock/runTypeComponentsStub.go [hunk 4]
```diff
@@ -20,6 +23,22 @@ func NewRunTypeComponentsStub() *RunTypeComponentsStub {
...
+		RequestHandlerFactory:            requestHandlers.NewResolverRequestHandlerFactory(),
+	}
+}
+
+// NewSovereignRunTypeComponentsStub -
+func NewSovereignRunTypeComponentsStub() *RunTypeComponentsStub {
+	runTypeComponents := NewRunTypeComponentsStub()
+
+	additionalStorageServiceCreator, _ := storageFactory.NewSovereignAdditionalStorageServiceFactory()
+	requestHandlerFactory, _ := requestHandlers.NewSovereignResolverRequestHandlerFactory(runTypeComponents.RequestHandlerFactory)
```
Comment (mariusmihaic) [inline epochStart/mock/runTypeComponentsStub.go:35]: Same remark, this is not a stub anymore...

File: factory/bootstrap/bootstrapComponents_test.go [hunk 5]
```diff
@@ -164,6 +164,21 @@ func TestNewBootstrapComponentsFactory(t *testing.T) {
 		require.Nil(t, bcf)
 		require.Equal(t, errorsMx.ErrNilNodesCoordinatorFactory, err)
 	})
+	t.Run("nil RequestHandlerFactory should error", func(t *testing.T) {
+		t.Parallel()
+
+		argsCopy := args
+		argsCopy.RunTypeComponents = &mainFactoryMocks.RunTypeComponentsStub{
```
Comment (mariusmihaic) [inline factory/bootstrap/bootstrapComponents_test.go:171]: Same remark for these tests

File: epochStart/mock/requestHandlerFactoryMock.go (hunk 3 above)
Comment (axenteoctavian) [reply epochStart/mock/requestHandlerFactoryMock.go:10]: deleted

File: epochStart/bootstrap/storageProcess_test.go (hunk 2 above)
Comment (axenteoctavian) [reply epochStart/bootstrap/storageProcess_test.go:60]: all the parameters are checked in `CreateEpochStartBootstrapper` function, and are covered by tests in `process_test.go`

**Review by raduchis — APPROVED** [review]

File: epochStart/mock/runTypeComponentsStub.go (hunk 4 above)
Comment (axenteoctavian) [reply epochStart/mock/runTypeComponentsStub.go:35]: it's not... but it will now work later if we will create tests for sovereign code coverage...

Comment (mariusmihaic) [reply epochStart/mock/runTypeComponentsStub.go:35]: When we'll create those type of tests, we can directly replace the needed mocks with real components. If tests work without real components, I would suggest we only use mocks for now

Comment (axenteoctavian) [reply epochStart/mock/runTypeComponentsStub.go]: changed to mocks where it was possible

**Review by mariusmihaic — DISMISSED** [review]

**Review by raduchis — APPROVED** [review]

**Review by mariusmihaic — APPROVED** [review]

**Pull request merged**